src/presets/firebase/preset.ts

```diff
@@ -33,6 +33,12 @@ const firebase = defineNitroPreset(
         nitro.options.appConfig.nitro = nitro.options.appConfig.nitro || {};
         nitro.options.appConfig.nitro.firebase = nitro.options.firebase;
 
+        if(nitro.options.firebase?.serverFunctionName.includes('-')){
```

Comment:
You can use [this util](https://github.com/unjs/knitwork#gensafevariablenamename) to check if name is not equal after normalizing, it is an issue.

Reply:
So you are suggesting no normalising just reject the build if `serverFunctionName` is not safe ?

Reply:
yes.